File: quint/prolog/lists/p05.pl
% P05 (*) Reverse a list.

my_reverse(L,K) :-
    acc_reverse(L,K,[]).

% Return the accumulator.
acc_reverse([],L,L).
% Continue with the tail and add the removed head to the front of the accumulator.
acc_reverse([X|Xs],L,Acc) :-
    acc_reverse(Xs,L,[X|Acc]).

File: quint/haskell/lists/p05.hs
-- P05 (*) Reverse a list.

my_reverse :: [x] -> [x]
my_reverse l = my_reverse_r l []

my_reverse_r :: [x] -> [x] -> [x]
my_reverse_r [] x     = x
my_reverse_r (x:xs) y = my_reverse_r xs ([x] ++ y)
